### ISSUE TITLE
No value returned from seetalert2, undefined

### ISSUE THREAD

**Issue by AlaeddineMessadi** [limonte/sweetalert2#41]
Let's say I have this code:

Confirm function
```
function confirmFn() {
return  swal({
            title: 'Confirmation',
            html: ' Are you sure? ',
            type: 'warning',
            showCancelButton: true,
            closeOnConfirm: true,
            closeOnCancel: true
        }, function (isConfirm) {
            if (isConfirm) {
                return true;
            } else {
                return false;
            }
        });
}
```
If you call the function, the return value should be true or false

```
var reternedValue = confirmFn() ;

alert(reternedValue);  // undefined
```
Then sweetalert popups correctly but it doesn't hold the script waiting for the user's action, confirm or cancel.

**Comment by limonte** [comment]
You should read and understand what are callbacks in JS: http://dreamerslab.com/blog/en/javascript-callbacks/

TLDR: `return  swal(…)` will always return `undefined` BEFORE `function (isConfirm) { … }` is even called. So you need to refactor you app to work correctly with swal2 callback.

**Comment by AlaeddineMessadi** [comment]
Thank you :)

**Comment by paschalidi** [comment]
Still dont really get how I should work with that . Can you please provide an example ?
Sorry for the stupid question :/

**Comment by birjolaxew** [comment]
@paschalidi This issue is *very* old, from back before the library moved to using Promises instead. The same basic problem still applies though - you're trying to use `swal({ ... })` as a synchronous function, while user input by definition is an asynchronous action.

Here's an example of how to work with the "new" Promise API:

```javascript
swal({ ... })
    .then(function(){
        alert("The confirm button was clicked");
    }).catch(function(reason){
        alert("The alert was dismissed by the user: "+reason);
    });
```

A bunch more examples are in the readme.

**Comment by paschalidi** [comment]
Thanks for the fast reply @birjolaxew  really appreciated :)

**Comment by paschalidi** [comment]
So you think it is possible to POST an ajax request when 
`.then(function (isConfirm) {
    //ajaxPOST()
}`

**Comment by limonte** [comment]
@paschalidi RTFM.

@birjolaxew don't waste your precious time answering such low-quality questions.

**Comment by birjolaxew** [comment]
On a closing note: this has little to do with SweetAlert2, and more to do with Promises. You should read up on those instead of asking here.
